mathesar_ui/src/stores/table-data/constraints.ts

```diff
@@ -95,6 +101,10 @@ export class ConstraintsDataStore implements Writable<ConstraintsData> {
     return null;
   }
 
+  drop(constraintId: Constraint['id']): CancellablePromise<undefined> {
+    return deleteAPI(`/tables/${this.parentId}/constraints/${constraintId}`);
+  }
```

Comment:
This function only performs an API call and does not update the store itself, which makes it inconsistent. We can follow the pattern we discussed to encapsulate the api calls into an object. The drop function can then call fetch within it, after the constraint is deleted.

Reply:
This would also remove the need for manipulating the loading state from within the component.

Reply:
Addressed in 087452ba0941b2690a0d20bc81d5025c0713a65c